src/journal/compress.c

```diff
@@ -677,12 +865,102 @@ int decompress_stream_lz4(int in, int out, uint64_t max_bytes) {
 #endif
 }
 
+int decompress_stream_zstd(int fdf, int fdt, uint64_t max_bytes) {
+#if HAVE_ZSTD
+        _cleanup_(ZSTD_freeDCtxp) ZSTD_DCtx *s = ZSTD_createDStream();
+        size_t out_size = ZSTD_CStreamOutSize();
+        uint8_t out[out_size];
```

Comment:
`out_size` here and in `compress_stream_zstd()` is about 128 KB. Is it okay to allocate that much on the stack?

Any size will work, so if you don't want to allocate a buffer, you could use a smaller buffer like 4 KB.